src/entities/communitySbt.ts

```diff
@@ -189,15 +224,167 @@ class SBT {
         try {
             await this.contract.callStatic.multiRedeem(data.leaves, data.proofs, data.roots);
             const tx = await this.contract.multiRedeem(data.leaves, data.proofs, data.roots);
-            await tx.wait()
             return {
-                claimedBadgeTypes
+                claimedBadgeTypes,
             }
         } catch (err) {
             throw new Error("Unable to claim multiple badges");
         }
     }
 
+    public async getSeasonBadges({
+        subgraphUrl,
+        dbUrl,
+        userId,
+        seasonId,
+      }: {
+        subgraphUrl: string;
+        dbUrl: string;
+        userId: string;
+        seasonId: number;
+      }): Promise<BadgeResponse[]> {
+        if (!subgraphUrl || !dbUrl) {
+          return [];
+        }
+        try {
+            const badgeQuery = `
+                query( $id: BigInt) {
+                    seasonUser(id: $id) {
+                        id
+                        badges {
+                          id
+                          awardedTimestamp
+                          mintedTimestamp
+                          badgeType
+                          badgeName
+                        }
+                    }
+                }
+            `;
+            const client = new ApolloClient({
+                cache: new InMemoryCache(),
+                link: new HttpLink({ uri: subgraphUrl, fetch })
+            })
+            const id = `${userId.toLowerCase()}#${seasonId}`
+            const data = await client.query({
+                query: gql(badgeQuery),
+                variables: {
+                    id: id,
+                },
+            });
+      
+            if (!data.data.seasonUser) {
+                return [];// empty array
+            }
+
+            const nonProgBadges = await this.getNonProgramaticBadges(userId, dbUrl);
+      
+            const subgraphBadges = data.data.seasonUser.badges as BadgeResponse[];
+            let badgesResponse : BadgeResponse[] = [];
```

Comment:
```suggestion
            let badgesResponse : BadgeResponse[] = badgesResponse.slice();
```

Reply:
logic has changed in 5f76461fb3a0e6766a31e4c37d0a913b7ea09481. I don't just need the copy now